apps/api/tasks.py

```diff
@@ -0,0 +1,58 @@
+import logging
+from uuid import UUID
+
+from celery.app import shared_task
+from django.contrib.contenttypes.models import ContentType
+from django.db.models import Subquery
+from taskbadger.celery import Task as TaskbadgerTask
+
+from apps.channels.clients.connect_client import CommCareConnectClient
+from apps.channels.models import ChannelPlatform, ExperimentChannel
+from apps.experiments.models import Experiment, ParticipantData
+
+logger = logging.getLogger(__name__)
+
+
+@shared_task(bind=True, base=TaskbadgerTask, ignore_result=True)
+def setup_connect_channels_for_bots(self, connect_id: UUID, experiment_data_map: dict):
+    """
+    Set up Connect channels for experiments that are using the ConnectMessaging channel
+
+    experiment_data_map: {experiment_id: participant_data_id}
+    """
+
+    experiment_ids = list(experiment_data_map.keys())
+    participant_data_ids = list(experiment_data_map.values())
+    connect_specific_data = ParticipantData.objects.filter(
+        id__in=participant_data_ids, participant__platform=ChannelPlatform.COMMCARE_CONNECT
```

Comment:
This filtering is already done in the view right? Do we need to do it again? Also, it seems to be repeated in the main query below.

Reply:
You are correct. I forgot to update this filter after I added it in the view, thanks

Reply:
https://github.com/dimagi/open-chat-studio/pull/1035/commits/a920c1a2ff0389aa5ff204dcb120ade85b94df9b